src/heretic/utils.py

```diff
@@ -121,3 +124,107 @@ def get_readme_intro(
 -----
 
 """
+
+
+def is_notebook() -> bool:
```

Comment:
Is there a library that does this? I'd rather not maintain such a brittle function myself.